distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestSingleSignOn.java

```diff
@@ -119,6 +120,25 @@ public String toString() {
         METADATA, PROTOCOL
     }
 
+    @Before
+    public void setup() throws Exception {
+        // Get all of the metadata
+        String metadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String ddfSpMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // Make sure all the metadata is valid before we set it
+        validateSaml(metadata, SamlSchema.METADATA);
+        validateSaml(ddfSpMetadata, SamlSchema.METADATA);
+
+
+        // The IdP server can point to multiple Service Providers and as such expects an array.
+        // Thus, even though we are only setting a single item, we must wrap it in an array.
+        setConfig("org.codice.ddf.security.idp.client.IdpMetadata", "metadata", metadata);
```

Comment:
This change, and the itests that were broken prior to making it, expose a secondary bug that was lurking in the old implementation. Nothing was clearing out old metadata from the internal maps in `IdpEndpoint` and `IdpMetadata` on re-injection of their associated lookup properties. That's resolved as well now.